### ISSUE TITLE
Either SA1009 or SA1024 raised for primary constructor with class inheritance

### ISSUE THREAD

**Issue by sobiyang** [DotNetAnalyzers/StyleCopAnalyzers#3775]
In C# 12, if we use primary constructor with class inheritance, then SA1009 is raised. However if we removed the trailing space, it becomes SA1024.

Sample code:
 ```cs
 public class ServiceOne(int intervalSeconds) : BackgroundService
    {
        protected override async Task ExecuteAsync(CancellationToken stoppingToken)
        {
            while (!stoppingToken.IsCancellationRequested)
            {
                await Task.Delay(intervalSeconds * 1000, stoppingToken);
            }
        }
    }
```
Keeping the space:
![StyleCop_SA1009](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/assets/3262885/002379c7-6142-4868-ae59-efe3197b5980)
Removing the space:
![StyleCop_SA1024](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/assets/3262885/071d9809-a038-4cad-a1cf-3e0c1fc95c2c)

**Comment by sharwell** [comment]
Duplicate of #3248
